### DISCUSSION TITLE
Questions about redirects

### DISCUSSION THREAD

**Discussion by forthrin** [HoneyryderChuck/httpx#40] [Q&A]
1. Is there a shorthand to get the final URL of a completed request? (eg. `HTTPX.get(url).url`) Feature request?
2. [Follow Redirects](https://honeyryderchuck.gitlab.io/httpx/wiki/Follow-Redirects) says to use `plugin(:follow_redirects)` which is deprecated. (The suggested `:callbacks` does work.)
3. The [event callbacks](https://honeyryderchuck.gitlab.io/httpx/wiki/Events) link on the above page gives a 404.

Bonus questions:

4. Is there an overview somewhere of all the methods (and public properties, if any) available on a response object?
5. `ri`documentation is basically non-existing for this library, at least for my installation. Isn't this auto-generated?

**Comment by HoneyryderChuck** [comment]
1. `HTTPX::Response` responds to `#uri`, which delegates to the corresponding request. This is smth which is not OOTB documented,  as rdoc ignores `def_delegator` callls, willl see about adding it manually.
2.  plugin(:follow_redirects) is not deprecated. Perhaps you mean that the callbacks section is out-of-date, I've just updated it [here](https://gitlab.com/os85/httpx/-/wikis/Follow-Redirects).
3. Also fixed.
4. https://honeyryderchuck.gitlab.io/httpx/rdoc/
5. not sure, haven't used ri outside of ruby core and stdlib docs. You probably have to do smth on your side for docs to be installed in the proper format, but if there's smth required from the gem, feel free to contribute.

**Comment by forthrin** [comment]
1. Indeed. Much obliged!
2. The code claims otherwise.
```
HTTPX.plugin(:follow_redirects).on_response_completed { |_, res| all << res }.get('https://www.github.com/')
DEPRECATION WARNING: calling `.on_response_completed` on plain HTTPX sessions is deprecated. Use HTTPX.plugin(:callbacks).on_response_completed instead.
```
3. Still 404.

> How can I get the sequence of redirect responses? You can us[e!] [event callbacks](https://honeyryderchuck.gitlab.io/httpx/wiki/Events) to collect them:

4. Brilliant!
5. I've not yet familiarized myself with how Ruby automates documentation. No biggie personally. Just FYI.

**Comment by forthrin** [comment]
Bump

**Comment by HoneyryderChuck** [comment]
2. code claims `on_response_completed` is deprecated, not the `:follow_redirects` plugin. [this announcement](https://honeyryderchuck.gitlab.io/httpx/rdoc/files/doc/release_notes/1_2_0_md.html#label-3Acallbacks+plugin) is helpful.
3. the plugins is [callbacks](https://honeyryderchuck.gitlab.io/httpx/wiki/Callbacks), not events. the link has been updated in the wiki pages, fwiw.
